### DISCUSSION TITLE
How do I handle Keyboard event, but only ones that doesn't produce ascii chararcters ?

### DISCUSSION THREAD

**Discussion by Maix0** [rust-windowing/winit#2604] [Q&A] [answered]
I want to handle keyboard event in some way for keys that produces ascii characters and another way for keys that don't.
I don't know how to do this, any ideas?

**Comment by kchibisov** [comment, marked as answer]
The `WindowEvent::KeyboardInput` is followed by the `WindowEvent::ReceivedCharacter(char)`. You can check that it's ascii with https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii.

**Reply by Maix0** [reply]
It seems like it the other way around sadly.

**Reply by madsmtm** [reply]
I think the order may be platform-dependent, then?

**Reply by kchibisov** [reply]
It must be KeyboardInput and then Character. If it's the other way around it's a bug.

**Reply by madsmtm** [reply]
@Maix0 could you file an issue about it if so (along with platform information)
